packages/react-ui-core/src/Gmap/Layer.js

```diff
@@ -0,0 +1,58 @@
+import { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+export default class Layer extends PureComponent {
+  static propTypes = {
+    map: PropTypes.object,
+    geojson: PropTypes.object.isRequired,
+    // styles that apply to line, polygon geometries
+    styles: PropTypes.shape({
+      clickable: PropTypes.boolean,
+      draggable: PropTypes.boolean,
+      editable: PropTypes.boolean,
+      fillColor: PropTypes.string,
+      fillOpacity: PropTypes.number,
+      strokeColor: PropTypes.string,
+      strokeOpacity: PropTypes.number,
+      strokeWeight: PropTypes.number,
+      visible: PropTypes.boolean,
+      zIndex: PropTypes.number,
+    }),
+  }
+
+  static defaultProps = {
+    styles: {},
+  }
+
+  constructor(props) {
+    super(props)
+    this.features = new Set()
+  }
+
+  componentWillUnmount() {
+    this.removeFeatures()
+  }
+
+  removeFeatures() {
+    const { map } = this.props
+
+    this.features.forEach(feature => {
+      if (map) map.data.remove(feature)
+      this.features.delete(feature)
```

Comment:
It's probably slightly more efficient to do `this.features.clear()` after the `forEach`. But if this set only has a few hundred items it probably doesn't matter much.

Reply:
If it's ever possible it would have more than that, I'd prefer the clear

Reply:
agreed, `clear()` is better, i'll update